modules/extension/xsd/xsd-core/src/main/java/org/geotools/xsd/StreamingParser.java

```diff
@@ -214,6 +215,10 @@ protected StreamingParser(
         this.input = input;
     }
 
+    public void setEntityResolver(EntityResolver entityResolver) {
```

Comment:
Looks good to me... while you're at it, do you want to add a similar method to the PullParser as well? (it's used by the WFS store).

Reply:
The PullParser uses a different XML API so I can completely disable external entities there but not set an entity resolver.  This shouldn't be a problem for the WFS store since its other feature parser that doesn't use PullParser already disables external entities.